XPathTest/Qt3Tests/Qt3Assertions.cs

```diff
@@ -363,31 +364,36 @@ static Qt3Assertions()
                 );
                 return (xpath, contextNode, variablesInScope, namespaceResolver) =>
                 {
-                    // var errorContents = "";
                     var actualErrorCode = "";
-                    Assert.Throws<XPathException>(
-                        () =>
-                        {
-                            try
-                            {
-                                Evaluate.EvaluateXPathToString(xpath,
-                                    contextNode,
-                                    domFacade,
-                                    new Options<TNode>(
-                                        namespaceResolver,
-                                        documentWriter: nodesFactory,
-                                        languageId: language),
-                                    variablesInScope);
-                            }
-                            catch (XPathException ex)
-                            {
-                                actualErrorCode = ex.ErrorCode;
-                                // errorContents = ex.Message;
-                                throw;
-                            }
-                        }
+                    var errorMessage = "";
+                    try
+                    {
+                        Evaluate.EvaluateXPathToString(xpath,
+                            contextNode,
+                            domFacade,
+                            new Options<TNode>(
+                                namespaceResolver,
+                                documentWriter: nodesFactory,
+                                languageId: language),
+                            variablesInScope);
+                    }
+                    catch (XPathException ex)
+                    {
+                        actualErrorCode = ex.ErrorCode;
+                    }
+                    catch (Exception ex)
+                    {
+                        // Hopefully extracts error code from non-XPathExceptions
+                        errorMessage = ex.Message;
```

Comment:
Do we have those? I'd expect all the XPath exceptions to be XPathExceptions, not just normal exceptions